cmd/config_helper.go

```diff
@@ -92,9 +118,19 @@ func collectProviders(pkgSets ...[]*api.Package) map[string]string {
 	return providers
 }
 
-func collectDependencies(pkg string, requires []string, providers map[string]string, ignored map[string]bool) ([]string, error) {
+func collectDependencies(pkg string, requires []string, providers map[string]string, ignored map[string]bool, allPackages map[string]*bazeldnf.RPM) ([]string, error) {
+	logrus.Debugf("Collecting dependencies for %s", pkg)
 	depSet := make(map[string]bool)
-	for _, req := range requires {
+	explored := make(map[string]bool)
+	for len(requires) > 0 {
```

Comment:
Can we just use something like https://pkg.go.dev/github.com/dominikbraun/graph and do a topological sort?